lib/src/soloud.dart

```diff
@@ -1432,6 +1432,84 @@ interface class SoLoud {
     return SoLoudController().soLoudFFI.getIsValidVoiceHandle(handle);
   }
 
+  /// Returns the number of concurrent sounds that are playing at the moment.
+  int getActiveVoiceCount() {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+    return SoLoudController().soLoudFFI.getActiveVoiceCount();
+  }
+
+  /// Returns the number of concurrent sounds that are playing a
+  /// specific audio source.
+  int getCountAudioSource(int soundHash) {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+    return SoLoudController().soLoudFFI.getCountAudioSource(soundHash);
+  }
+
+  /// Returns the number of voices the application has told SoLoud to play.
+  int getVoiceCount() {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+    return SoLoudController().soLoudFFI.getVoiceCount();
+  }
+
+  // /// Get a sound's protection state.
+  // bool getProtectVoice(int handle) {
+  //   if (!isInitialized) {
+  //     throw const SoLoudNotInitializedException();
+  //   }
+  //   return SoLoudController().soLoudFFI.getProtectVoice(handle);
+  // }
+
+  // /// Set a sound's protection state.
+  // ///
+  // /// Normally, if you try to play more sounds than there are voices,
+  // /// SoLoud will kill off the oldest playing sound to make room.
+  // /// This will most likely be your background music. This can be worked
+  // /// around by protecting the sound.
+  // /// If all voices are protected, the result will be undefined.
+  // ///
+  // /// [handle]  handle to check.
+  // /// [protect] whether to protect or not.
+  // void setProtectVoice(int handle, bool protect) {
+  //   if (!isInitialized) {
+  //     throw const SoLoudNotInitializedException();
+  //   }
+  //   SoLoudController().soLoudFFI.setProtectVoice(handle, protect);
+  // }
```

Comment:
Why are these commented out?

Reply:
I wrote that in the description. It seems that `setProtectVoice` does nothing as stated in [this issue](https://github.com/jarikomppa/soloud/issues/298). I tried that fix, but it doesn't work for me. I have to do some more research and tests.

Anyway `setMaxActiveVoiceCount` does its job. The default max voices are set to 16 by default but can be raised to 4095. Maybe 16 is a low max number for games, we may think about raising it when initializing the player or let the devs set it while calling `initialize`.

`setProtectVoice()` just tells the player to continue to play that sound even if the max active voice count is reached (ie a background music). The default value of 16 is for performance. The more concurrent voices are playing, the more CPU power is needed. So it's also up to devs to check this.

This is explained better [here](https://solhsa.com/soloud/soloud_20200207.html#voice).